src/components/sequencing/actions/Actions.svelte

```diff
@@ -0,0 +1,531 @@
+    workspaceActionRuns = Object.values($actionRunsByWorkspace[workspaceId] || {});
+  }
+
+  $: selectedActionRuns = (workspaceActionRuns || []).filter(actionRun => {
+    return actionRun.action_definition_id === selectedActionDefinition?.id;
+  });
+
+  $: filteredActionRuns = (selectedActionDefinition ? selectedActionRuns : workspaceActionRuns).filter(actionRun => {
+    const definition = getActionDefinitionForRun(actionRun, $actionDefinitionsByWorkspace, workspaceId);
+    if (definition && definition.name.indexOf(actionRunsFilterText) > -1) {
+      return true;
+    }
+    if (actionRun.created_by && actionRun.created_by.indexOf(actionRunsFilterText) > -1) {
+      return true;
+    }
+    return false;
+  });
+
+  $: filteredActionDefinitions = workspaceActionDefinitions.filter(actionDefinition => {
+    return actionDefinition.name.indexOf(actionDefinitionsFilterText) > -1;
+  });
+
+  $: if (typeof selectedActionDefinitionId === 'number') {
+    selectedActionDefinition =
+      workspaceActionDefinitions.find(actionDefinition => actionDefinition.id === selectedActionDefinitionId) || null;
+  } else {
+    selectedActionDefinition = null;
+  }
+
+  $: if (selectedActionDefinition) {
+    name = selectedActionDefinition.name;
+    description = selectedActionDefinition.description;
+    argumentsMap = selectedActionDefinition.settings;
+    getCode(selectedActionDefinition.action_file_id, user);
+  } else {
+    name = '';
+    description = '';
+    argumentsMap = {};
+  }
+
+  $: saveButtonDisabled = !name;
+
+  onMount(() => {
+    workspaceId = getSearchParameterNumber(SearchParameters.WORKSPACE_ID);
+  });
+
+  async function getCode(fileId: number, user: User | null) {
+    if (selectedActionDefinition) {
+      if (codeAbortController) {
+        codeAbortController.abort();
+      }
+      codeAbortController = new AbortController();
+      code = '';
+      const { aborted, file } = await effects.getFile(fileId, user, codeAbortController.signal);
+      if (!aborted && typeof file === 'string') {
+        code = file;
+      }
+    } else {
+      code = '';
+    }
+  }
+
+  async function onNewActionClick() {
+    if (typeof workspaceId !== 'number') {
+      return;
+    }
+
+    showActionCreationModal(user, workspaceId);
+  }
+
+  function onActionRunClick(id: number) {
+    const workspaceId = getSearchParameterNumber(SearchParameters.WORKSPACE_ID);
+    goto(
+      `${base}/sequencing/actions/runs/${id}${workspaceId ? `?${SearchParameters.WORKSPACE_ID}=${workspaceId}` : ''}`,
+    );
+  }
+
+  async function runAction(action: ActionDefinition) {
+    const actionRunId = await effects.runAction(action, user);
+    if (typeof actionRunId === 'number') {
+      goto(
+        `${base}/sequencing/actions/runs/${actionRunId}${workspaceId ? `?${SearchParameters.WORKSPACE_ID}=${workspaceId}` : ''}`,
+      );
+    }
+  }
+
+  function getActionDefinitionForRun(
```

Comment:
This function was already defined in the `ActionRun.svelte` component. Would it be possible to break it out into a reusable util function?